sfft_and_animate.py

```diff
@@ -1,300 +1,308 @@
+
+        return None, None
+
+def animate(frames,labels,oid,band,savename,artist='Lauren Aldoroty',savedir='/hpc/group/cosmology/lna18/roman_sim_imgs/Roman_Rubin_Sims_2024/gifs/'):
+    """
+    Animate a series of frames. 
+    """
+    # NOTE: I still need to align the images to the same WCS for the animations. 
+    if not os.path.exists(savedir):
+        os.mkdir(savedir)
     
-    metadata = dict(artist='Lauren Aldoroty')
-    animate_stamps(animate_panels, savepath, metadata, labels=animate_pointings, staticlabel=band)
+    savepath = os.path.join(savedir,savename)
+    metadata = dict(artist=artist)
+
+    animate_stamps(frames,savepath,metadata,labels,staticlabel=band)
+
+def run(oid,band,n_templates=1,verbose=False):
+
+    ra,dec,start,end = set_sn(oid)
+    in_tab,out_tab = sn_in_or_out(oid,start,end,band)
+
+    template_tab = out_tab[:n_templates]
+    if verbose:
+        print('The template images are:')
+        print(template_tab)
+
+    for t_row in template_tab:
+        t_pointing, t_sca = t_row['pointing'], t_row['sca']
+
+        dd_stamps = []
+        skysub_stamps = []
+        pointings = []
+        for row in in_tab:
+            print('*************************************************************')
+            sci_pointing, sci_sca = row['pointing'], row['sca']
+            print(band,sci_pointing,sci_sca,'-',band,t_pointing,t_sca)
+
+            skysubimgpath, ddimgpath = sfft(ra,dec,band,sci_pointing,sci_sca,t_pointing,t_sca,verbose=verbose)
+            if skysubimgpath is not None:
+                # Make stamps.
+                dd_stamp_savepath = f'/work/lna18/imsub_out/decorr/stamps/dd_stamp_{oid}_{band}_{sci_pointing}_{sci_sca}_-_{t_pointing}_{t_sca}.fits'
+                skysub_stamp_savepath = f'/work/lna18/imsub_out/skysub/stamps/skysub_stamp_{oid}_{band}_{sci_pointing}_{sci_sca}_-_{t_pointing}_{t_sca}.fits'
+
+                ddstamppath = stampmaker(ra,dec,ddimgpath,savepath=dd_stamp_savepath,shape=np.array([100,100]))
+                skysubstamppath = stampmaker(ra,dec,ddimgpath,savepath=skysub_stamp_savepath,shape=np.array([100,100]))
+            
+                # Prepare information for animations.
+                pointings.append(sci_pointing)
+                with fits.open(ddstamppath) as ddhdu:
+                    ddimg = ddhdu[0].data
+                    dd_stamps.append(ddimg)
+
+                with fits.open(skysubstamppath) as skysubhdu:
+                    skysubimg = skysubhdu[0].data
+                    skysub_stamps.append(skysubimg)
+
+            # Make animations.
+            # They are saved with the SN ID, followed by the image used as a template.
+            animate(dd_stamps,pointings,oid,band,f'{oid}/{oid}_{band}_{t_pointing}_{t_sca}_SFFT.gif')
+
+    # I don't need N_templates animations for the sky subtracted-only images, so this is out of the loop.
+    animate(skysub_stamps,pointings,oid,band,f'{oid}/{oid}_{band}_RAW.gif')
+
+def parse_slurm():
```

Comment:
We should be able to share this between `mk_lc.py` and `sfft_and_animate`.py

Reply:
I think eventually it will be just one pipeline file when it's all smoothed out--I think this would ultimately be most efficient once we are confident it works correctly.